tests/test_replay.py

```diff
@@ -270,3 +271,62 @@ def test_filter_out_tests_not_in_file(testdir):
         ],
         consecutive=True,
     )
+
+
+def test_replay_file_outcome_is_correct(testdir):
+    """Tests that the outcomes in the replay file are correct."""
+    testdir.makepyfile(
+        test_module="""
+        def test_success():
+            pass
+
+        def test_failure():
+            assert False
+    """
+    )
+    dir = testdir.tmpdir / "replay"
+    result = testdir.runpytest_subprocess(f"--replay-record-dir={dir}")
+    assert result.ret != 0
+
+    contents = [json.loads(s) for s in (dir / ".pytest-replay.txt").read().splitlines()]
+    assert len(contents) == 4
+
+    assert "test_success" in contents[1]["nodeid"]
+    assert contents[1]["outcome"] == "passed"
+
+    assert "test_failure" in contents[3]["nodeid"]
+    assert contents[3]["outcome"] == "failed"
```

Comment:
```suggestion
    testdir.makepyfile(
        test_module="""
        import pytest
        
        def test_success():
            pass

        def test_failure():
            assert False
            
        @pytest.fixture
        def some_fixture():
            yield
            assert False
            
        def test_failure_fixture_teardown(some_fixture):
            assert True
    """
    )
    dir = testdir.tmpdir / "replay"
    result = testdir.runpytest_subprocess(f"--replay-record-dir={dir}")
    assert result.ret != 0

    contents = [json.loads(s) for s in (dir / ".pytest-replay.txt").read().splitlines()]
    outcomes = {r["nodeid"]: r["outcome"] for r in contents if "outcome" in r}
    assert outcomes == {
        "test_module.py::test_success": "passed",
        "test_module.py::test_failure": "failed",
        "test_module.py::test_failure_fixture_teardown": "failed",
    }
```
Updating to test my previous suggestion.

Reply:
this misses a test for a setup failure, adding it